Race condition in `util.Queue.Dequeue()` causes panic: index out of range [0]

# Scrapligo Race Condition Bug Report

## Description

There is a race condition in the `Dequeue()` method of `util/queue.go` that can cause a panic when multiple goroutines are accessing the queue concurrently.

### The Problem

The current implementation checks the queue depth before acquiring the lock, but doesn't verify the queue is still non-empty after acquiring the lock:

```go
func (q *Queue) Dequeue() []byte {
    // check the depth before acquiring a full read/write lock which can cause deadlocks with tons
    // of concurrent access to enqueue/deque.
    if q.getDepth() == 0 {
        return nil
    }

    q.lock.Lock()           // ← Another goroutine can empty the queue HERE
    defer q.lock.Unlock()

    b := q.queue[0]         // ← PANIC: index out of range [0] with length 0
    // ...
}
```

### Race Condition Scenario

1. Goroutine A checks `getDepth()` and sees depth > 0
2. Goroutine B acquires the lock and dequeues the last item
3. Goroutine A acquires the lock and tries to access `q.queue[0]`
4. **PANIC**: `runtime error: index out of range [0] with length 0`

### Stack Trace Example

```
panic: runtime error: index out of range [0] with length 0

goroutine 887 [running]:
github.com/scrapli/scrapligo/util.(*Queue).Dequeue(0xc00126f7a0)
    /path/to/vendor/github.com/scrapli/scrapligo/util/queue.go:64 +0x1b0
github.com/scrapli/scrapligo/channel.(*Channel).Read(0xc001249860)
    /path/to/vendor/github.com/scrapli/scrapligo/channel/read.go:129 +0x4f
github.com/scrapli/scrapligo/channel.(*Channel).ReadUntilPrompt(0xc001249860, {0x28e2898, 0xc000b55030})
    /path/to/vendor/github.com/scrapli/scrapligo/channel/read.go:234 +0x95
```

### Impact

This issue occurs under high concurrency when multiple SSH connections are being managed simultaneously. It causes the application to crash unexpectedly.

### Environment

- scrapligo version: v1.3.3
- Go version: 1.23.8
- OS: Linux (Docker container)
- Concurrency: 50 worker goroutines managing SSH connections

### Reproduction

This race condition is more likely to occur with:
- High number of concurrent SSH connections
- Fast connect/disconnect cycles
- Multiple goroutines reading from the same channel

## Proposed Solution

Add a double-check after acquiring the lock to ensure the queue is still non-empty:

```go
func (q *Queue) Dequeue() []byte {
    if q.getDepth() == 0 {
        return nil
    }

    q.lock.Lock()
    defer q.lock.Unlock()

    // Double-check after acquiring lock to prevent race condition
    if len(q.queue) == 0 {
        return nil
    }

    b := q.queue[0]

    q.queue = q.queue[1:]
    q.depth--

    <-q.depthChan
    q.depthChan <- q.depth

    return b
}
```

This follows the "check-lock-check" pattern commonly used to prevent TOCTOU (Time-of-check to time-of-use) race conditions.

## Verification

After applying this fix, our production environment with 50 concurrent workers has been running without panics for [duration].

## Additional Notes

The same pattern could potentially be applied to `DequeueAll()` and other queue methods for consistency, though we haven't observed panics in those methods yet.

--- a/util/queue.go
+++ b/util/queue.go
@@ -61,6 +61,11 @@ func (q *Queue) Dequeue() []byte {
 	q.lock.Lock()
 	defer q.lock.Unlock()
 
+	// Double-check after acquiring lock to prevent race condition
+	if len(q.queue) == 0 {
+		return nil
+	}
+
 	b := q.queue[0]
 
 	q.queue = q.queue[1:]

